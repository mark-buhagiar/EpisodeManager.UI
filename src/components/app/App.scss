.app {
    display: flex;
    flex-flow: column;
    min-height: 100vh;
}

.app-container {
    margin-left: auto;
    margin-right: auto;
    flex: 1 1 auto;
    position: relative;
}

.no-select {
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -khtml-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

@media (min-width: 1200px) {
    .app-container {
        width: 1170px;
    }
}

@media (min-width: 992px) and (max-width: 1199px) {
    .app-container {
        width: 970px;
    }
}

@media (min-width: 768px) and (max-width: 991px) {
    .app-container {
        width: 750px;
    }
}
