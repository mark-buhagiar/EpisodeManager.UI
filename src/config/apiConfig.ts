import EnvironmentConfig from './environmentConfig';

let ApiConfig = {
    feedParserApiBase: 'https://feedparser.episodemanager.eu/api/v1/',
    showsApiBase: 'https://shows.episodemanager.eu/api/v1/',
    usersApiBase: 'https://users.episodemanager.eu/api/v1/',
};

if (EnvironmentConfig.isMocking) {
    ApiConfig = {
        feedParserApiBase: 'http://localhost:3001/',
        showsApiBase: 'http://localhost:3001/',
        usersApiBase: 'http://localhost:3001/',
    };
}

export default ApiConfig;
