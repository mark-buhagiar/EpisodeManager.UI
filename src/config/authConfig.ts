import EnvironmentConfig from './environmentConfig';

const AuthConfig = {
    domain: 'episodemanager.eu.auth0.com',
    clientId: 'jho2qfvcVXxe30tA7fZepmfLTfZizj1W',
    audience: 'https://episodemanager.eu',
    get permissionsClaim(): string {
        return this.audience + '/permissions';
    },
};

/* use the test client on Auth0
AuthConfig = {
    ...AuthConfig,
    clientId: '9JEzUu9Ka1TmfaqjAAUIIoc8aAHqEKdR',
};
*/

export default AuthConfig;
