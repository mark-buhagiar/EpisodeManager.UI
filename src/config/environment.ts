const EnvironmentConfig = {
    isMocking: false,
    isAuthenticated: false,
};

export default EnvironmentConfig;
