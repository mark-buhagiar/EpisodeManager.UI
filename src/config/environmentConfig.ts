const EnvironmentConfig = {
    isMocking: true,
    isAuthenticated: true,
    loading: false, // Only used when isAuthed is set to false
    permissions: ['read:admin', 'execute:admin'],
};

export default EnvironmentConfig;
