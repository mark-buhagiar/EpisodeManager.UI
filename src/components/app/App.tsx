import React from 'react';
import { withAuth } from '../../HoC/withAuth';
import withAuthHeaderInterceptor from '../../HoC/withAuthHeaderInterceptor';
import Header from '../header/Header';
import Footer from '../footer/Footer';
import Routes from '../routes';
import './App.scss';
import { createMuiTheme, ThemeProvider } from '@material-ui/core';

const theme = createMuiTheme({
    palette: {
        type: 'dark',
        primary: {
            main: '#e2e2e2',
        },
        text: {
            primary: '#e2e2e2',
            secondary: '#e2e2e2',
        },
        action: {
            active: '#e2e2e2',
            hover: '#333333',
        },
        background: {
            paper: '#2a292b',
        },
    },
});

const App = (): JSX.Element => {
    return (
        <ThemeProvider theme={theme}>
            <div className="app">
                <Header />
                <div className="app-container">
                    <Routes />
                </div>
                <Footer />
            </div>
        </ThemeProvider>
    );
};

export default withAuth(withAuthHeaderInterceptor(App));
