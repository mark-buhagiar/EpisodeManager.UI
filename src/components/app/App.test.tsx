import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import App from './App';
import * as environmentConfig from '../../config/environmentConfig';

describe('When accessing the site', () => {
    beforeAll(() => {
        environmentConfig.default.isMocking = true;
    });

    it.each`
        test              | isAuthenticated | isLoading
        ${'loading page'} | ${false}        | ${true}
        ${'landing page'} | ${false}        | ${false}
        ${'home page'}    | ${true}         | ${false}
    `(
        'should render the $test if authentication is $isAuthenticated and loading is $isLoading',
        ({ test, isAuthenticated, isLoading }): void => {
            environmentConfig.default.isAuthenticated = isAuthenticated;
            environmentConfig.default.loading = isLoading;

            const application = TestRenderer.create(
                <MemoryRouter>
                    <App />
                </MemoryRouter>,
            );
            expect(application).toBeTruthy();
            expect(application).toMatchSnapshot();
        },
    );
});
