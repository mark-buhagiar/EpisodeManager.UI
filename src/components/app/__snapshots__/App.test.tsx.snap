// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When accessing the site should render the home page if authentication is true and loading is false 1`] = `
<div
  className="app"
>
  <nav
    className="nav-bar no-select"
  >
    <div
      className="app-container"
      id="navContainer"
    >
      <span
        className="app-title"
      >
        Episode Manager
      </span>
      <a
        aria-current="page"
        className="nav-link active-nav-link"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Calendar
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/shows"
        onClick={[Function]}
      >
        Shows
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/help"
        onClick={[Function]}
      >
        Help
      </a>
      <span
        className="spacer"
      />
      <a
        aria-current={null}
        className="nav-link"
        href="/profile"
        onClick={[Function]}
      >
        Profile
      </a>
      <span
        className="nav-link"
      >
        Log Out
      </span>
    </div>
  </nav>
  <div
    className="Toastify"
  />
  <div
    className="app-container"
  >
    <div
      className="homepage"
    >
      <div
        className="message"
      >
        Hello! Welcome to Episode Manager 2.0. Everything here is very much still a work in progress, so if you run into any issues or have any feedback, please get in touch 
        <a
          href="mailto:markbuh25@gmail.com"
        >
          here
        </a>
        . To set up your shows, please visit your profile page. If you need help remembering what you were subscribed to, get in touch and I will forward you your list üòä.
      </div>
      <div
        className="calendar-navigation"
      >
        <div
          className="qualities-legend"
        >
          <div>
            <div
              className="quality-indicator standard-def"
              title="Standard definition"
            />
             
            Standard definition
          </div>
          <div>
            <div
              className="quality-indicator p720"
              title="720p"
            />
             
            720p
          </div>
          <div>
            <div
              className="quality-indicator p1080"
              title="1080p"
            />
             
            1080p
          </div>
        </div>
        <div
          className="date-selector"
        >
          <div
            className="date-changer no-select "
            id="navigate_before"
            onClick={[Function]}
          >
            <span
              className="material-icons"
            >
              navigate_before
            </span>
          </div>
          <div
            className="month-label"
          >
            May
             
            <br />
             
            <div
              className="year"
            >
              2020
            </div>
          </div>
          <div
            className="date-changer no-select "
            id="navigate_next"
            onClick={[Function]}
          >
            <span
              className="material-icons"
            >
              navigate_next
            </span>
          </div>
        </div>
        <div
          className="button-group refresh-button"
        >
          <button
            className="button no-select "
            onClick={[Function]}
            title=""
            type="button"
          >
            
            Refresh
          </button>
        </div>
      </div>
      <div
        className="calendar"
      >
        <div
          className="calendar-header"
        >
          <div
            className="calendar-header-item"
          >
            Mon
          </div>
          <div
            className="calendar-header-item"
          >
            Tue
          </div>
          <div
            className="calendar-header-item"
          >
            Wed
          </div>
          <div
            className="calendar-header-item"
          >
            Thu
          </div>
          <div
            className="calendar-header-item"
          >
            Fri
          </div>
          <div
            className="calendar-header-item"
          >
            Sat
          </div>
          <div
            className="calendar-header-item"
          >
            Sun
          </div>
        </div>
        <div
          className="calendar-body"
        />
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <a
      href="https://www.linkedin.com/in/buhagiar-mark/"
      target="blank"
    >
      From Mark with
       
      <span
        aria-label="heart"
        role="img"
      >
        ‚ô•Ô∏è
      </span>
       
      - 
      2020
    </a>
  </div>
</div>
`;

exports[`When accessing the site should render the landing page if authentication is false and loading is false 1`] = `
<div
  className="app"
>
  <nav
    className="nav-bar no-select"
  >
    <div
      className="app-container"
      id="navContainer"
    >
      <span
        className="app-title"
      >
        Episode Manager
      </span>
      <span
        className="spacer"
      />
      <span
        className="nav-link"
      >
        Log In
      </span>
    </div>
  </nav>
  <div
    className="Toastify"
  />
  <div
    className="app-container"
  >
    <div
      className="landing-page"
    >
      <div
        className="title site-name"
      >
        Episode
        <br />
        Manager
        <span
          className="version-number"
        >
          v2.0
        </span>
      </div>
      <div
        className="button-group button-group"
      >
        <button
          className="button no-select "
          onClick={[Function]}
          title=""
          type="button"
        >
          
          Log In / Sign Up
        </button>
      </div>
      <div
        className="sub-text"
      >
        Please contact me if you previously had an account and would like your data to be migrated
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <a
      href="https://www.linkedin.com/in/buhagiar-mark/"
      target="blank"
    >
      From Mark with
       
      <span
        aria-label="heart"
        role="img"
      >
        ‚ô•Ô∏è
      </span>
       
      - 
      2020
    </a>
  </div>
</div>
`;

exports[`When accessing the site should render the loading page if authentication is false and loading is true 1`] = `
<div
  className="app"
>
  <nav
    className="nav-bar no-select"
  >
    <div
      className="app-container"
      id="navContainer"
    >
      <span
        className="app-title"
      >
        Episode Manager
      </span>
      <span
        className="spacer"
      />
      <span
        className="nav-link"
      >
        Log In
      </span>
    </div>
  </nav>
  <div
    className="Toastify"
  />
  <div
    className="app-container"
  >
    <div
      className="loading-container"
    >
      <div
        className="loading-indicator"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <a
      href="https://www.linkedin.com/in/buhagiar-mark/"
      target="blank"
    >
      From Mark with
       
      <span
        aria-label="heart"
        role="img"
      >
        ‚ô•Ô∏è
      </span>
       
      - 
      2020
    </a>
  </div>
</div>
`;
